docs/guides/security/basics/what-is-cryptography/index.md

```diff
@@ -0,0 +1,105 @@
+
+Of these goals, confidentiality carries the most weight. The need to ensure that an unauthorized party cannot access the data is the ultimate objective of cryptography. That does not mean that the remaining goals are of less importance.
+
+Data integrity is vital to ensure that the message has not been altered in some way. Otherwise, the receiving party could be manipulated into taking a wrong or undesirable action. Whether a spy is sending a message to their country’s leadership, or a company is sending instructions to a field office, both sender and receiver need assurance that the message sent is identical to the message received.
+
+Authenticity is essential to ensure that the user or system are known and trusted. Establishing the identity of the user (sender or recipient) is the crux of this assurance. However, the system must also be known in order to [prevent ransomware attacks](/docs/guides/ransomware-attack) that involve phishing (fraudulent emails), vishing (fraudulent voice mails and phone calls), smishing (fraudulent texts), and other deceptive forms of communication.
+
+## Types of Cryptography
+
+There are three types of cryptography: secret key cryptography, public key cryptography, and hash functions.
+
+The least complicated and fastest to use is secret key cryptography, also known as symmetric cryptography. This type uses one key to encrypt and decrypt communications. It protects data at rest and data in transit, and is most often used on data at rest. The most well-known algorithms used in secret key cryptography are [Advanced Encryption Standard (AES)](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard), [Triple Data Encryption Standard (3DES)](https://en.wikipedia.org/wiki/Triple_DES), and [Rivest Cipher 4 (RC4)](https://en.wikipedia.org/wiki/RC4).
+
+Public key cryptography, or asymmetric cryptography, uses two keys on each end of the communication. Each pair consists of a public and a private key. Public keys are exchanged between sender and recipient. The sender then uses the recipient’s public key to encrypt the message. The recipient uses their private key to decrypt the message. Examples of public key use are plentiful in just about any communication over the Internet such as [HTTPS](/docs/guides/introducing-http-2/), [SSH](/docs/guides/connect-to-server-over-ssh-on-linux/), [OpenPGP](https://www.openpgp.org/), [S/MIME](https://en.wikipedia.org/wiki/S/MIME), and a [website’s SSL/TLS certificate](/docs/guides/what-is-a-tls-certificate/).
+
+The math connecting public and private keys makes it impossible to derive the private key from the public key. The public key is derived from the private key. This is why private keys should never be shared.
```

Comment:
The math connecting public and private keys makes it impossible to derive the private key from the public key. However, the public key is derived from the private key, which is why private keys should never be shared.